### PR TITLE
Dotnetcore

### PR DESCRIPTION
**Pull request by TrueUO**
DO NOT MERGE

- We are keeping this PR open until it is confirmed stable on a few shards. We really appreciate any feedback or PRs from the community on this one.

### CODE REVIEW

**Comment by deccer** [comment]
Edit: Turned my comment into a review.

**Comment by TrueUO** [comment]
> Ahoy there.. looks like `Application\ServUO.csproj` has the wrong OutDirs set.
> 
> ```
> 	<PublishDir>..\..\Distribution</PublishDir>
> 	<OutDir>..\..\Distribution</OutDir>
> ```
> 
> should be
> 
> ```
> 	<PublishDir>..\Distribution</PublishDir>
> 	<OutDir>..\Distribution</OutDir>
> ```

@Argalep

File: Application/ServUO.csproj [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
-    <OutputPath>..\</OutputPath>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <ConcurrentGarbageCollection>true</ConcurrentGarbageCollection>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <StartupObject>ServUO.ServUO</StartupObject>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Comment (deccer) [inline Application/ServUO.csproj]: Those should be `..\Distribution` instead

File: Scripts/Scripts.csproj [hunk 2]
```diff
@@ -2,14 +2,16 @@
...
+    <TargetFramework>netcoreapp3.1</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <ConcurrentGarbageCollection>true</ConcurrentGarbageCollection>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Comment (deccer) [inline Scripts/Scripts.csproj]: Those should be `..\Distribution` instead

File: Server/Server.csproj [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-    <TargetFramework>net48</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <ConcurrentGarbageCollection>true</ConcurrentGarbageCollection>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Comment (deccer) [inline Server/Server.csproj]: Those should be `..\Distribution` instead

File: Ultima/Ultima.csproj [hunk 4]
```diff
@@ -2,14 +2,16 @@
...
+	<TargetFramework>netcoreapp3.1</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
     <ServerGarbageCollection>true</ServerGarbageCollection>
     <ConcurrentGarbageCollection>true</ConcurrentGarbageCollection>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RootNamespace>Server</RootNamespace>
+	<PublishDir>..\..\Distribution</PublishDir>
+	<OutDir>..\..\Distribution</OutDir>
```
Comment (deccer) [inline Ultima/Ultima.csproj]: Those should be `..\Distribution` instead

**Review by deccer — CHANGES_REQUESTED** [review]
I decided to do it properly, and turn the comment into a review. I hope you don't mind.

File: Ultima/Ultima.csproj (hunk 4 above)
Comment (TrueUO) [reply Ultima/Ultima.csproj]: Thanks a lot for the help @deccer!

**Comment by kevin-10** [comment]
When archived saves are set to false, none of that code is even called. Are we absolutely sure that's what was causing the extra save time?

**Comment by TrueUO** [comment]
> When archived saves are set to false, none of that code is even called. Are we absolutely sure that's what was causing the extra save time?

Even archivedsaves are off, it does still seem to check if it is off, during each save causing the new delay when trying to make it compatible.

When I reverted to the old method it worked again, no 2x save issues. It has to do with when we tried to update it recently.

**Comment by TrueUO** [comment]
@Argalep has a potential fix just waiting for him to push it.

**Comment by TrueUO** [comment]
I pushed @Argalep suggestion and re-added @Vorspire original changes to ArchivedSaves. 

Have not yet tested on Heritage.

**Comment by TrueUO** [comment]
Just closing this for now due to holding up checks on the main. Is it a sonar issue?

**Pull request closed without merging**